Civi/API/Subscriber/I18nSubscriber.php

```diff
@@ -98,38 +98,22 @@ public function onApiRespond(\Civi\API\Event\Event $event) {
    * Sets the tsLocale and dbLocale for multi-lingual sites.
    * Some code duplication from CRM/Core/BAO/ConfigSetting.php retrieve()
    * to avoid regressions from refactoring.
-   * @param $lcMessagesRequest
+   * @param string $lcMessages
    * @param int $requestId
    * @throws \API_Exception
    */
-  public function setLocale($lcMessagesRequest, $requestId) {
-    // We must validate whether the locale is valid, otherwise setting a bad
-    // dbLocale could probably lead to sql-injection.
+  public function setLocale($lcMessages, $requestId) {
     $domain = new \CRM_Core_DAO_Domain();
     $domain->id = \CRM_Core_Config::domainID();
     $domain->find(TRUE);
 
-    // are we in a multi-language setup?
-    $multiLang = $domain->locales ? TRUE : FALSE;
-    $lcMessages = NULL;
-
-    // on multi-lang sites based on request and civicrm_uf_match
-    if ($multiLang) {
-      $config = \CRM_Core_Config::singleton();
-      $languageLimit = [];
-      if (isset($config->languageLimit) and $config->languageLimit) {
-        $languageLimit = $config->languageLimit;
-      }
-
-      if (in_array($lcMessagesRequest, array_keys($languageLimit))) {
```

Comment:
This is the reason the test was passing even though it was setting `languageLimit` incorrectly. Instead of doing a simple `array_key_exists` check, this was doing it a roundabout way by flipping the keys and then checking with `in_array`, which left it vulnerable to the [infamous php bug where `in_array` will match the number `0` to any string](https://stackoverflow.com/questions/13846769/php-in-array-0-value). So this validation was essentially broken.